ImputationPipeline/ManualQCPRS/ManualQCPRS.py

```diff
@@ -0,0 +1,884 @@
+import ipywidgets as widgets
+from IPython.display import display
+from IPython.display import clear_output
+import pandas as pd
+import firecloud.api as fapi
+import json
+from google.cloud import storage
+from datetime import datetime
+import pytz
+from abc import ABC, abstractmethod
+import argparse
+
+
+def get_bucket_and_blob(uri):
+    return uri.replace("gs://", "").split("/", 1)
+
+
+class color:
+    PURPLE = '\033[95m'
+    CYAN = '\033[96m'
+    DARKCYAN = '\033[36m'
+    BLUE = '\033[94m'
+    GREEN = '\033[92m'
+    YELLOW = '\033[93m'
+    RED = '\033[91m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+    END = '\033[0m'
+
+
+class TableSelectionGUI:
+    def __init__(self, workspace_namespace, workspace_name, workspace_bucket_name, available_tables):
+        self.available_tables = available_tables
+        self.workspace_namespace = workspace_namespace
+        self.workspace_name = workspace_name
+        self.workspace_bucket_name = workspace_bucket_name
+
+    def run(self):
+        self.build_table_selection_dropdown()
+
+    def build_table_selection_dropdown(self):
+        self.table_selection_dropdown = widgets.Dropdown(options=[None] + self.available_tables,
+                                                         description='Select Table to Load Batches From',
+                                                         style={"description_width": 'initial'},
+                                                         layout=widgets.Layout(width='auto')
+                                                         )
+        self.table_selection_dropdown.observe(self.select_table, names="value")
+        display(self.table_selection_dropdown)
+
+    def select_table(self, change):
+        if change['old'] != None:
+            self.app.close_widgets()
```

Comment:
My IDE complains that `self.app` isn't defined, and I also don't immediately see where it would be. It seems to work though, can you maybe quickly double-check that this is working as intended?